internal/lock.go

```diff
@@ -26,178 +28,295 @@ type config struct {
 	Resource []Resource
 }
 
+// NewLock creates a new lock file instance
 func NewLock(path string, newOk bool) (*Lock, error) {
 	_, error := os.Stat(path)
 	if os.IsNotExist(error) {
 		if newOk {
-			return &Lock{path: path}, nil
+			return &Lock{path: path, conf: config{Resource: []Resource{}}}, nil
 		} else {
 			return nil, fmt.Errorf("file '%s' does not exist", path)
 		}
 	}
+
 	var conf config
 	file, err := os.Open(path)
 	if err != nil {
-		return nil, error
+		return nil, err
```

Comment:
This looks like a legit bug fix!

But again: it should be in a separate PR -- not part of this PR.